src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -602,6 +605,9 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
                 block = BindLambdaBody(lambdaSymbol, lambdaBodyBinder, diagnostics);
             }
 
+            // PROTOTYPE: Test that we're reporting diagnostics even in the case where we re-used the lambda above.
+            lambdaSymbol.GetDeclarationDiagnostics(diagnostics);
```

Comment:
Why are we not using `AddDeclarationDiagnostics` here, which will ensure diagnostics are actually calculated?

Reply:
> Why are we not using `AddDeclarationDiagnostics` here, which will ensure diagnostics are actually calculated?

`AddDeclarationDiagnostics()` only adds diagnostics that have already been calculated. `GetDeclarationDiagnostics()` will ensure diagnostics are added to `_declarationDiagnostics`.

---
In reply to: [605833428](https://github.com/dotnet/roslyn/pull/52178#discussion_r605833428) [](ancestors = 605833428)